### PR TITLE
feat: add 11-brushy-cli exploration for user+artwork pair

### PR DESCRIPTION
**Pull request by anandaroop**
https://www.notion.so/artsy/AI-Innovation-Squad-Q2-2024-619cf183deb449e39cea411777af5071?p=83087024f4d44d5d866fb4664fb3ada4&pm=s

This PR adds experiment no. 11 — whose purpose is to rough out a data approach for the idea of an assistant that kicks in on an artwork page to tell a story about why that artwork might fit a particular collector.

This adds [commander.js](https://github.com/tj/commander.js) to the project to simplify the acceptance of command line args, e.g.:

```sh
 yarn tsx src/11-brushy-cli/01-artwork --user 63dbf037d98f90000dabcdef --artwork 665088d0eaf0d1125babcdef
```

That command will:

- Take the specified artwork and run it through a computer vision step to generate a brief prose description
- Take that^ along with a selection of metadata re: user, artwork, artist, gallery and send it to the LLM for a response

The response ideally relates the artwork in question to the user's interests and collection.

Results were shared here:

https://docs.google.com/spreadsheets/d/11TT765nI_NlS6viNjemR2dVhICVQKJcUBBUxAomjIc8/edit#gid=79995578

### CODE REVIEW

File: src/11-brushy-cli/data.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+  // console.log(JSON.stringify(data, null, 2))
+
+  function cleanUp(names: { name: string }[]) {
+    return names.map((x) => x.name).join("; ")
+  }
+
+  // clean up user - follows
+  const followGenes = extractNodes(data.user.follows.genesConnection)
+  data.user.followedCategories = cleanUp(followGenes)
+  delete data.user.follows.genesConnection
```
[the remark this answers is not in the document]
Comment (mc-jones) [reply src/11-brushy-cli/data.ts:115]: Curious if you tested it both ways and saw improved output from the cleaned up `JSON`?

File: src/11-brushy-cli/data.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    {},
+    { "X-Access-Token": process.env.ADMIN_ACCESS_TOKEN! }
+  )
+  artwork.categories = Object.keys(pacs).join("; ")
+
+  // console.log({ artwork, partner, artist, user, collectedArtworks })
+  return { artwork, partner, artist, user, collectedArtworks }
+}
+
+export async function getPartnerAppliedGene(name: string) {
```
Comment (mc-jones) [inline src/11-brushy-cli/data.ts:163]: Don't have to resolve in this PR, but just wonder out loud if we should extract out these functions into the `tools` folder as this data will likely be useful for a variety of use cases?

**Review by mc-jones — APPROVED** [review]
Very nice. Few questions/comments, but looks great. 🙌

File: src/11-brushy-cli/data.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (anandaroop) [reply src/11-brushy-cli/data.ts:115]: I presumed that fewer tokens / less noise is better, but I didn't actually test that systematically.

File: src/11-brushy-cli/data.ts (hunk 2 above)
Comment (anandaroop) [reply src/11-brushy-cli/data.ts:163]: Ah interesting point. Probably so, I'll fold that into my tools cleanup PR (which is looking more like next week at this point)

**Pull request merged**